apps/api/src/app/environments/usecases/get-api-rate-limit/get-api-rate-limit.usecase.ts

```diff
@@ -0,0 +1,77 @@
+import { Injectable, InternalServerErrorException, Logger } from '@nestjs/common';
+import { EnvironmentRepository, OrganizationRepository } from '@novu/dal';
+import { buildApiRateLimitKey, CachedEntity } from '@novu/application-generic';
+import { ApiRateLimitCategoryTypeEnum, ApiServiceLevelTypeEnum, IApiRateLimits } from '@novu/shared';
+import { GetApiRateLimitCommand } from './get-api-rate-limit.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+import { GetDefaultApiRateLimits } from '../../../rate-limiting/usecases/get-default-api-rate-limits';
+
+const LOG_CONTEXT = 'GetApiRateLimit';
+
+@Injectable()
+export class GetApiRateLimit {
+  constructor(
+    private environmentRepository: EnvironmentRepository,
+    private organizationRespository: OrganizationRepository,
+    private getDefaultApiRateLimits: GetDefaultApiRateLimits
+  ) {}
+
+  async execute(command: GetApiRateLimitCommand): Promise<number> {
+    return await this.getApiRateLimit({
+      apiRateLimitCategory: command.apiRateLimitCategory,
+      _environmentId: command.environmentId,
+      _organizationId: command.organizationId,
+    });
+  }
+
+  @CachedEntity({
+    builder: (command: GetApiRateLimitCommand) =>
+      buildApiRateLimitKey({
+        _environmentId: command.environmentId,
+        apiRateLimitCategory: command.apiRateLimitCategory,
+      }),
+  })
```

Comment:
Caching the result of the API rate limit lookup for a given Environment and API Rate Limit category to improve performance. This will be used by the token bucket refresh algorithm.